src/main/java/ch/njol/skript/bukkitutil/sounds/SoundReceiver.java

```diff
@@ -0,0 +1,132 @@
+package ch.njol.skript.bukkitutil.sounds;
+
+import ch.njol.skript.Skript;
+import org.bukkit.Location;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Sound;
+import org.bukkit.SoundCategory;
+import org.bukkit.World;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+
+import java.util.Locale;
+import java.util.OptionalLong;
+
+/**
+ * Adapter pattern to unify {@link World} and {@link Player} playSound methods.
+ * Methods can be called without determining version support, it is handled internally.
+ * Non-supported methods will simply delegate to supported methods.
+ */
+public interface SoundReceiver {
+
+	boolean ADVENTURE_API = Skript.classExists("net.kyori.adventure.sound.Sound$Builder");
+	boolean SPIGOT_SOUND_SEED = Skript.methodExists(Player.class, "playSound", Entity.class, Sound.class, SoundCategory.class, float.class, float.class, long.class);
+	boolean ENTITY_EMITTER_SOUND = Skript.methodExists(Player.class, "playSound", Entity.class, Sound.class, SoundCategory.class, float.class, float.class);
+	boolean ENTITY_EMITTER_STRING = Skript.methodExists(Player.class, "playSound", Entity.class, String.class, SoundCategory.class, float.class, float.class);
```

Comment:
Static so multiple extending interfaces don't reinitialize heavy reflection checks.
```suggestion
	static boolean ADVENTURE_API = Skript.classExists("net.kyori.adventure.sound.Sound$Builder");
	static boolean SPIGOT_SOUND_SEED = Skript.methodExists(Player.class, "playSound", Entity.class, Sound.class, SoundCategory.class, float.class, float.class, long.class);
	static boolean ENTITY_EMITTER_SOUND = Skript.methodExists(Player.class, "playSound", Entity.class, Sound.class, SoundCategory.class, float.class, float.class);
	static boolean ENTITY_EMITTER_STRING = Skript.methodExists(Player.class, "playSound", Entity.class, String.class, SoundCategory.class, float.class, float.class);
```

Reply:
They already are static, you can't have non-static interface fields.

Reply:
The exact reason to keep the static definition

Reply:
lime adding static doesn't do anything but pop up an ide inspection saying that you should remove it. there's no point to adding the keyword static here.